### ISSUE TITLE
Clarification on GHSA-chcr-x7hc-8fp8

### ISSUE THREAD

**Issue by G-Rath** [devise-two-factor/devise-two-factor#264]
Is there an aspect of GHSA-chcr-x7hc-8fp8 that is specific to `devise-two-factor`, or is it just that it has no built in mechanism to counter brute force attacks?

If it's the latter I wonder if the security advisory could be withdrawn (while keeping the readme update) as that feels out of scope of this library and having it as an advisory results in _any_ use of this library being flagged by security monitors with the only course of action to ignore it (which for me comes with a bunch of paper work that has to be carried throughout the applications lifecycle).

**Comment by jeffreygray** [comment]
+1 to this

tagging @bsedat to hear your thoughts whenever you have a sec!

**Comment by bsedat** [comment]
There is nothing specific to devise-two-factor. The purpose of the CVE and GHSA is to make sure users are aware that they need to implement mitigations for the risk. 
Apologies for the paperwork. I'd like to learn more about it. What workflows are flagging the advisory?

**Comment by G-Rath** [comment]
Any consumer of the GitHub Advisory Database, which are most of the big players - so dependabot:

<img width="710" alt="image" src="https://github.com/devise-two-factor/devise-two-factor/assets/3151613/fd23d413-21cc-43e6-88ac-9b3411744258">

And tools like [`osv-detector`](https://github.com/G-Rath/osv-detector) and [`osv-scanner`](https://github.com/google/osv-scanner) which use https://osv.dev/ which aggregates from GHADB among other sources:

```
❯ osv-detector .
Loaded the following OSV databases:
  RubyGems (802 vulnerabilities, including withdrawn - last updated Tue, 16 Jan 2024 17:31:37 GMT)

Gemfile.lock: found 225 packages
  Using db RubyGems (802 vulnerabilities, including withdrawn - last updated Tue, 16 Jan 2024 17:31:37 GMT)

  devise-two-factor@5.0.0 is affected by the following vulnerabilities:
    GHSA-chcr-x7hc-8fp8: Devise-Two-Factor vulnerable to brute force attacks (https://github.com/advisories/GHSA-chcr-x7hc-8fp8)

  1 known vulnerability found in Gemfile.lock
```

Those are the tools I use at my work, though I would assume other tools like synk would also be flagging this

**Comment by bsedat** [comment]
Thank you for the information. I'll bring it internally and discuss with the team - removing or disabling the devise-two-factor gem like bundler-audit suggests would be would be worse for security overall.

More of my thoughts for additional clarity: devise-two-factor doesn't natively protect against brute force attacks but applications using it for 2FA still have a better security posture than ones without. A attacker would need knowledge of the existing username, password, and get lucky with a collision. Without a 2FA solution like devise-two-factor the user account would already have been compromised with the known username/password.

**Comment by dan-jensen** [comment]
For anyone using `bundler-audit`, you can ignore this CVE using the ["ignore" syntax](https://github.com/rubysec/bundler-audit#readme) like this: `bundle-audit check --ignore CVE-2024-0227`.

**Comment by G-Rath** [comment]
@bsedat any update on this? I appreciate what you're trying to do with this but it's not really a good fit with how advisories work.

Personally I think the update on the readme about this is really the way to go.

**Comment by bsedat** [comment]
Hi everyone,
Thanks for your patience. We're going to be moving forward by retracting the CVE and GHSA. The underlying issue is out of scope for the project and a problem with the underlying protocol but we'd still like to communicate the issue to make sure it gets addressed at the application level. I'll report here once that process is done.
I'm sorry for the inconvenience this caused!

**Comment by G-Rath** [comment]
I'm going to resolve this since the advisory has now been withdrawn
